File: 2025/Company/Company.Web/Modules/Company/BusinessEntity/BusinessEntityForm.cs
namespace Company.Company.Forms;

[FormScript("Company.BusinessEntity")]
[BasedOnRow(typeof(BusinessEntityRow), CheckNames = true)]
public class BusinessEntityForm
{
    [Category("Person")]
    //public int BusinessEntityId { get; set; }
    [OneThirdWidth]
    public string PersonType { get; set; }
    [OneThirdWidth]
    public string Title { get; set; }
    [OneThirdWidth]
    public string Suffix { get; set; }
    [OneThirdWidth]
    public string FirstName { get; set; }
    [OneThirdWidth]
    public string MiddleName { get; set; }
    [OneThirdWidth]
    public string LastName { get; set; }

    [Category("Addresses")]
    public List<BusinessEntityRow> AddressList { get; set; }

    [HalfWidth]
    public bool IsActive { get; set; }

}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/BusinessEntityForm.ts
import { StringEditor, BooleanEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface BusinessEntityForm {
    PersonType: StringEditor;
    Title: StringEditor;
    Suffix: StringEditor;
    FirstName: StringEditor;
    MiddleName: StringEditor;
    LastName: StringEditor;
    AddressList: StringEditor;
    IsActive: BooleanEditor;
}

export class BusinessEntityForm extends PrefixedContext {
    static readonly formKey = 'Company.BusinessEntity';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!BusinessEntityForm.init)  {
            BusinessEntityForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;

            initFormType(BusinessEntityForm, [
                'PersonType', w0,
                'Title', w0,
                'Suffix', w0,
                'FirstName', w0,
                'MiddleName', w0,
                'LastName', w0,
                'AddressList', w0,
                'IsActive', w1
            ]);
        }
    }
}